inst/scripts/.importMapCoord.R

```diff
@@ -0,0 +1,169 @@
+# importMapCoord.R
+# non-exported
+
+#==========================makeMap and Validation===============================
+
+# Helper function - generates the closure for a single gene model
+makeMap <- function(ucscMod) {
+
+    # Check if the number of exons listed in start and end match
+    if(length(ucscMod$exonStart) != length(ucscMod$exonEnd)) {
+        return(NULL)
+    }
+
+    # Prepare the environment (exon matrix, start, end, strand)
+    exons <- cbind(ucscMod$exonStart, ucscMod$exonEnd)
+    start <- ucscMod$start
+    end <- ucscMod$end
+    reverse <- FALSE
+    if (ucscMod$strand == "-") {
+        reverse <- TRUE
+    }
+
+    # Check if the exons and start stop points are validly ordered
+    if(is.unsorted(c(start, as.vector(t(exons)), end))) {
+        return(NULL)
+    }
+
+    # Return a function that uses the exon matrix, strand and start, stop positions
```

Comment:
How about pre-calculating a third column with cumulative exon-length sums  ```cumsum()``` so you don't need to do the sums in every function call. You could just then just use ```exons$cumsum[iEx] - (exons$end[iEx] - x)``` (probably need to add or subtract a 1 somewhere in there) to get the position. Also you could switch start and end for (-) strand, when you prepare ```exons```, then you don't have to special-case the (-)-strand.